packages/xstate-fsm/src/index.ts

```diff
@@ -231,7 +236,10 @@ export function interpret<
         unsubscribe: () => listeners.delete(listener)
       };
     },
-    start: () => {
+    start: (initialState: TState['value']) => {
+      if (initialState && initialState in machine.config.states) {
```

Comment:
Not sure if we need to second part of this condition. We either should not it at all or warn/throw (preferably only in dev) when this condition is met. Swallowing a possible mistake on the developer's side might cause discovering an error longer than needed.

Reply:
I like the idea to warn! 

There will be a JS error when trying to get `fsmConfig.states[state].entry` if the provided state does not exist in the configuration